readme.md

```diff
@@ -0,0 +1,105 @@
+# Intro
+
+This is MVP for one of our client (UTSEUS) which helps teams on hackathons exchange skills among team members and pay for their help.  It has three modules: web client (React), server (NodeJS) and storage with some business logic on blockchain (ethereum). This doc descriebs how to setup environment and run server backed by blockchian. 
+
+## Install environment
+
+A whole environment is described in scripts of package.json. Almost all what you need is run this scripts from the server folder. 
+
+```
+npm run install
+npm run testrpc
+npm run truffle
+# run server API tests
+npm run test-mocha
+# run unit tests
+npm run test-jest
```

Comment:
failing tests:

```
rngadam@rngadam-mac ~/c/s/s/server> npm run test-jest

> utseus_api@1.0.0 test-jest /Users/rngadam/coderbunker/src/skills-market/server
> jest

 PASS  test/parser.test.js
  ● Console

    console.log parser/parser.js:56
      TYPE: bytes
    console.log parser/parser.js:32
      bytes : 6461766500000000000000000000000000000000000000000000000000000000
    console.log parser/parser.js:32
      bool : 0000000000000000000000000000000000000000000000000000000000000001
    console.log parser/parser.js:56
      TYPE: uint256[]
    console.log parser/parser.js:32
      uint256[] : 0000000000000000000000000000000000000000000000000000000000000001
    console.log parser/parser.js:32
      uint256[] : 0000000000000000000000000000000000000000000000000000000000000002
    console.log parser/parser.js:32
      uint256[] : 0000000000000000000000000000000000000000000000000000000000000003

 FAIL  test/app.test.js
  ● Test NodeJS API › Request /users - return 46 users

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      10 |         var urlUsers = url + "/users";
      11 |         request(urlUsers, function(error, response, body) {
    > 12 |             var json = JSON.parse(body);
      13 |             expect(JSON.parse(json.users.length)).to.equal(46);
      14 |             done();
      15 |         });
      
      at Request._callback (test/app.test.js:12:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /skills - return 30 skills

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      19 |         var urlSkills = url + "/skills";
      20 |         request(urlSkills, function(error, response, body) {
    > 21 |             var json = JSON.parse(body);
      22 |             expect(json.length).to.equal(14);
      23 |             done();
      24 |         });
      
      at Request._callback (test/app.test.js:21:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /register - return tx hash

    AssertionError: expected undefined to equal 'user was successfully registered'

      28 |         var urlRegister = url + "/register/Ricky/React";
      29 |         request.post(urlRegister, function(error, response, body) {
    > 30 |             expect(body).to.equal('user was successfully registered');
      31 |             done();
      32 |         })
      33 |     });
      
      at Request._callback (test/app.test.js:30:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /help - return positive response

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      36 |         var urlAskForHelp = url + "/help/Dmitry/React/10";
      37 |         request.post(urlAskForHelp, function(error, response, body) {
    > 38 |             var json = JSON.parse(body);
      39 |             expect(json.success).is.equal(true);
      40 |             expect(json.payload.length).not.equal(0);
      41 |             done();
      
      at Request._callback (test/app.test.js:38:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /help - return negative response

    AssertionError: expected undefined to equal 'User is not valid'

      46 |         var urlAskForHelp = url + "/help/Joe/React/10";
      47 |         request.post(urlAskForHelp, function(error, response, body) {
    > 48 |             expect(body).equal("User is not valid");
      49 |             done();
      50 |         })
      51 |     });
      
      at Request._callback (test/app.test.js:48:26)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /mentoring - mentee with request is exist - return positive response

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      57 |             var urlMentoring = url + "/mentoring/Ricky/Dmitry/React/10";
      58 |             request.post(urlMentoring, function(error, response, body) {
    > 59 |                 var json = JSON.parse(body);
      60 |                 expect(json.success).is.equal(true);
      61 |                 expect(json.payload.length).not.equal(0);
      62 |                 done();
      
      at Request._callback (test/app.test.js:59:33)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /mentoring - mentee with request isn't exist - return negative response

    AssertionError: expected undefined to equal 'User is not registered'

      68 |         var urlMentoring = url + "/mentoring/Ricky/Joe/React/10";
      69 |         request.post(urlMentoring, function(error, response, body) {
    > 70 |             expect(body).equal("User is not registered");
      71 |             done();
      72 |         });
      73 |     });
      
      at Request._callback (test/app.test.js:70:26)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /validate - positive scenario - return receipt

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      88 |         var urlAskForHelp = url + "/help/Dmitry/React/10";
      89 |         request.post(urlAskForHelp, function(error, response, body) {
    > 90 |             var json = JSON.parse(body);
      91 |             var urlValidate = url + "/validate/" + json.payload;
      92 |             request.get(urlValidate, function(error, response, body) {
      93 |                 var json = JSON.parse(body);
      
      at Request._callback (test/app.test.js:90:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

  ● Test NodeJS API › Request /validate - negative scenario - return error message

    SyntaxError: Unexpected token u in JSON at position 0
        at JSON.parse (<anonymous>)

      103 |         var urlValidate = url + "/validate/0x37049";
      104 |         request.get(urlValidate, function(error, response, body) {
    > 105 |             var json = JSON.parse(body);
      106 |             expect(json.success).is.equal(false);
      107 |             done();
      108 |         });
      
      at Request._callback (test/app.test.js:105:29)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.Object.<anonymous>.Request.onRequestError (node_modules/request/request.js:878:8)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       9 failed, 9 passed, 18 total
Snapshots:   0 total
Time:        4.906s
```

Reply:
Tests should be run when both blockhain server and NodeJs server are running. I move tests section after section which runs the server